### ISSUE TITLE
Possible regression in master: RuntimeError: call stack exhausted

### ISSUE THREAD

**Issue by mejedi** [wasmerio/wasmer#1598]
### Describe the bug

```sh
echo "`wasmer -V` | `rustc -V` | `uname -m`"
wasmer 0.17.1 | rustc 1.46.0 (04488afe3 2020-08-24) | x86_64
```
A binary that used to run just fine with the latest Wasmer release fails with `RuntimeError: call stack exhausted` if I build Wasmer from the current master.

```sh
$ (cd ~/wasmer.git && git describe --tags)
0.17.1-2043-gd7893fe5b
```

### Steps to reproduce

Save the following code to `reproduce.sh`. 
```sh
#!/bin/sh
set -eu
if [ ! -f llc.wasm ]; then wget https://registry-cdn.wapm.io/contents/rapidlua/llc/0.0.2/llc.wasm; fi
${WASMER:-wasmer} run --backend=cranelift llc.wasm <<EOF
%i4 = type <4 x i32>
%i8 = type <2 x i64>

declare %i4 @llvm.bswap.v4i32(%i4)
declare %i8 @llvm.bswap.v2i64(%i8)

declare %i4 @llvm.cttz.v4i32(%i4)
declare %i8 @llvm.cttz.v2i64(%i8)

define void @test_scalarized_intrinsics() {
        %r1 = add %i8 undef, undef

        %r2 = call %i4 @llvm.bswap.v4i32(%i4 undef)
        %r3 = call %i8 @llvm.bswap.v2i64(%i8 undef)

        %r4 = call %i4 @llvm.cttz.v4i32(%i4 undef)
        %r5 = call %i8 @llvm.cttz.v2i64(%i8 undef)

        ret void
}
EOF
```
The script pulls the binary from WAPM and invokes it with an input that triggers the error.

### Expected behavior

Works just fine with Wasmer latest release:
```sh
$ ./reproduce.sh
	.text
	.file	"<stdin>"
	.globl	test_scalarized_intrinsics      # -- Begin function test_scalarized_intrinsics
	.p2align	4, 0x90
	.type	test_scalarized_intrinsics,@function
test_scalarized_intrinsics:             # @test_scalarized_intrinsics
	.cfi_startproc
# %bb.0:
	retq
.Lfunc_end0:
	.size	test_scalarized_intrinsics, .Lfunc_end0-test_scalarized_intrinsics
	.cfi_endproc
                                        # -- End function
	.section	".note.GNU-stack","",@progbits
```

### Actual behavior
Fails with ` RuntimeError: call stack exhausted` when using Wasmer current master:
```sh
$ WASMER=~/wasmer.git/target/release/wasmer ./reproduce.sh
warning: the `--backend=cranelift` flag is deprecated, please use `--cranelift` instead
	.text
	.file	"<stdin>"
error: failed to run `llc.wasm`
│   1: WASI execution failed
│   2: failed to run WASI `_start` function
│   3: RuntimeError: call stack exhausted
           at <unnamed> (llc.wasm[18370]:0xffffffff)
           at <unnamed> (llc.wasm[5659]:0x3f8fb2)
           at <unnamed> (llc.wasm[4215]:0x2d16ba)
           at <unnamed> (llc.wasm[1779]:0x130587)
           at <unnamed> (llc.wasm[8559]:0x68823e)
           at <unnamed> (llc.wasm[17551]:0xc260aa)
           at <unnamed> (llc.wasm[8560]:0x688944)
           at <unnamed> (llc.wasm[17979]:0xc9d1cc)
           at <unnamed> (llc.wasm[11834]:0x92499e)
           at <unnamed> (llc.wasm[4399]:0x2ed600)
           at <unnamed> (llc.wasm[20871]:0xf36a68)
           at <unnamed> (llc.wasm[20885]:0xf37dda)
           at <unnamed> (llc.wasm[9666]:0x77574b)
           at <unnamed> (llc.wasm[13877]:0xa4195c)
           at <unnamed> (llc.wasm[14043]:0xa58699)
           at <unnamed> (llc.wasm[14524]:0xa96657)
           at <unnamed> (llc.wasm[14932]:0xaac409)
╰─> 4: stk_ovf
```

**Comment by mejedi** [comment]
Still observed in `1.0.0-alpha3`

**Comment by syrusakbary** [comment]
Hi @mejedi, thanks for opening the issue.

The issue should be now fixed thanks to the changes in #1167 :) (we tested and the `reproduce.sh` script works as expected)
